Ginger/Unit Tests/NonUITests/SourceControlUnitTest.cs

```diff
@@ -0,0 +1,155 @@
+
+        [TestMethod]
+        [DynamicData(nameof(GetData), DynamicDataSourceType.Method)]
+        public void GetProjectList(string SourceControlURL, string SourceControlUser, string SourceControlPass, string SourceControlType)
+        {
+            //Arrange
+            TestConnection(SourceControlURL, SourceControlUser, SourceControlPass, SourceControlType);
+            ObservableList<SolutionInfo> SourceControlSolutions = new ObservableList<SolutionInfo>();
+
+            //Act
+            SourceControlSolutions = SourceControl.GetProjectsList();
+
+            //Assert
+            Assert.AreEqual(SourceControlSolutions.Count, 1);
+        }
+        private static void DeleteDirectory(string directory)
+        {
+            foreach (string subdirectory in Directory.EnumerateDirectories(directory))
+            {
+                DeleteDirectory(subdirectory);
+            }
+
+            foreach (string fileName in Directory.EnumerateFiles(directory))
+            {
+                var fileInfo = new FileInfo(fileName)
+                {
+                    Attributes = FileAttributes.Normal
+                };
+                fileInfo.Delete();
+            }
+
+            Directory.Delete(directory);
+        }
+
+        [TestMethod]
+        [DynamicData(nameof(GetData), DynamicDataSourceType.Method)]
+        public void GetProject(string SourceControlURL, string SourceControlUser, string SourceControlPass, string SourceControlType)
+        {
+
+            //Arrange
+            string workingDirectory = Path.Combine(SolutionFolder, SourceControlType);
+
+            DeleteDirectory(workingDirectory);
+           
+            TestConnection(SourceControlURL, SourceControlUser, SourceControlPass, SourceControlType);
+            string error = string.Empty;
+            bool getproj = false;
+
+            //Act
+            getproj = SourceControl.GetProject(workingDirectory, SourceControlURL, ref error);
+
+            //Assert
+            Assert.AreEqual(getproj, true);
+        }
+
+        [TestMethod]
+        [DynamicData(nameof(GetData), DynamicDataSourceType.Method)]
+        public void CheckIn(string SourceControlURL, string SourceControlUser, string SourceControlPass, string SourceControlType)
+        {
+            //Arrange
+            TestConnection(SourceControlURL, SourceControlUser, SourceControlPass, SourceControlType);
+            GetProject(SourceControlURL, SourceControlUser, SourceControlPass, SourceControlType);
+            string workingDirectory = Path.Combine(SolutionFolder, SourceControlType);
+
+            File.WriteAllText(workingDirectory + "/Dummy.txt", File.ReadAllText(Path.Combine(workingDirectory, "Dummy.txt")).Replace("123", "1234"));
+            string error = string.Empty;
+            bool checkin = false;
+
+            //Act
+            checkin = SourceControl.CommitChanges("aaa", ref error);
+
+            //Assert
+            Assert.AreEqual(checkin, false);
```

Comment:
checkin = false means the checking passed?

Reply:
no , I have changed in the code